test/model/src/core/kv_transaction_snapshot.cpp

```diff
@@ -0,0 +1,69 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "model/kv_transaction_snapshot.h"
+#include "model/kv_update.h"
+
+namespace model {
+
+/*
+ * kv_transaction_snapshot_by_exclusion::contains --
+ *     Check whether the given update belongs to the snapshot.
+ */
+bool
+kv_transaction_snapshot_by_exclusion::contains(const kv_update &update) const noexcept
+{
+    if (update.txn_id() > _exclude_after)
+        return false;
+    return _exclude_ids.find(update.txn_id()) == _exclude_ids.end();
+}
+
+/*
+ * kv_transaction_snapshot_wt::contains --
+ *     Check whether the given update belongs to the snapshot.
```

Comment:
Can you please clarify what needs to be updated? The comment feels accurate to me...

Reply:
Sorry, it could just be me not reading the code carefully enough during the first pass.

After re-reading the code, one uses `_exclude_ids` and the other one `_include_ids`. I am wondering why we need those two implementations. I can see `kv_transaction_snapshot_by_exclusion` being used [here](https://github.com/wiredtiger/wiredtiger/pull/9883/files#diff-d9d563bdd0ddb81aab16c77526acd4c71da81d8767e64326bd4deef161202a6cR172) and there are functions such as `kv_database::txn_snapshot_nolock` that takes a `do_not_exclude` argument. If the intent is to create a snapshot and list the active transactions, why do we need that exclude/include boolean?

Reply:
Thank you for the explanation, I understand better now.